### PR TITLE
More advanced auth

### PR DESCRIPTION
**Pull request by dscottboggs**

### CODE REVIEW

**Comment by j8r** [comment]
Reborn of https://github.com/DFabric/dppm-rest-api/pull/2

File: src/dppm_rest_api/actions/app.cr [hunk 1]
```diff
@@ -1,4 +1,7 @@
 require "../../utils"
+require "dppm"
+require "../../config/route"
+require "../../utils"
```
Comment (j8r) [inline src/dppm_rest_api/actions/app.cr]: You shouldn't backward require modules, that the purpose of organizing our code in trees.
This simplify testing and refactoring, especially when changing paths

File: config.con [hunk 2]
```diff
@@ -1,5 +1,3 @@
 host "[::1]"
 port 8994
-sources {
-  default "https://github.com/DFabric/packages-source/tarball/master"
-}
+source "https://github.com/DFabric/packages-source/tarball/master"
```
Comment (j8r) [inline config.con]: the last one was correct, this change is in DPPM@master

Comment (dscottboggs) [reply config.con]: ok thanks I was wondering which of those it was

Comment (dscottboggs) [reply config.con]: I changed it back to `sources {...}` and even after running `shards update` I got an exception -- `Unhandled exception: Unexpected token: EndHash (Exception)`

Comment (j8r) [reply config.con]: Use this one https://github.com/DFabric/dppm/blob/master/config.con.
I also made a new commit to bump to Crystal 0.28.0, which is now in Alpine

Comment (dscottboggs) [reply config.con]: Thank you!

**Comment by dscottboggs** [comment]
CI is still failing due to #4.

**Comment by j8r** [comment]
No @dscottboggs , that's because the CI is now on Crystal 0.28.0. I've pushed a commit to fix the URI changes.

**Comment by dscottboggs** [comment]
I see. I have updated, but the CI is still failing due to a missing file. I'm not sure what that file is or why it's causing an exception.

**Review by j8r — APPROVED** [review]

**Pull request merged**